net/golioth/settings.c

```diff
@@ -93,6 +94,28 @@ static int send_status_report(struct golioth_client *client,
 	LOG_HEXDUMP_DBG(response_buf, nbytes, "Response");
 
 	return send_coap_response(client, response_buf, nbytes);
+#endif
+
+	uint8_t response_buf[GOLIOTH_SETTINGS_MAX_RESPONSE_LEN + 1] = {};
```

Comment:
I don't think there is a reason for `+ 1` in this context. We are creating binary data (CBOR), so initializing also seems unnecessary.